src/hvm.rs

```diff
@@ -982,6 +983,61 @@ impl FromStr for Name {
 impl U120 {
   pub const ZERO: U120 = U120(0);
   pub const MAX: U120 = U120(2_u128.pow(120) - 1);
+
+  pub fn wrapping_add(self, other:U120) -> U120 {
+    let res = self.0 + other.0;
+    U120(res & U120::MAX.0)
+  }
+  
+  pub fn wrapping_sub(self, other: U120) -> U120 {
+    let other_complement = U120::wrapping_add(U120(other.0 ^ U120::MAX.0), U120(1));
+    U120::wrapping_add(self, other_complement)
+  }
+
+  // based off of this answer https://stackoverflow.com/a/1815371
+  // maybe this is too much work for an easy function?
+  // idk, maybe there's a better way to do this
+  pub fn wrapping_mul(self, other: U120) -> U120 {
+    const LO_MASK : u128  =  (1 << 60) - 1;
+    let a = self.0;
+    let b = other.0;
+    let x = (a & LO_MASK) * (b & LO_MASK);
+    
+    let s0 = x & LO_MASK;
+    let x = (a >> 60) * (b & LO_MASK) + (x >> 60);
+    
+    let s1 = x & LO_MASK;
+    let s2 = x >> 60;
+    
+    let x = s2 + (a >> 60) * (b >> 60) + (x >> 60);
+    U120(s1 << 60 | s0)
```

Comment:
Don't know if more efficient, but I think more readable:
```suggestion
    let a_lo = a & LO_MASK;
    let a_hi = a >> 60;
    let b_lo = b & LO_MASK;
    let b_hi = b >> 60;
    s0 = a_lo * b_lo;
    s1 = ((a_hi * b_lo) & LO_MASK) << 60;
    s2 = ((b_hi * a_lo) & LO_MASK) << 60);
    U120(s0).wrapping_add(U120(s1)).wrapping_add(U120(s2));
```

Reply:
I agree.